elements/lisk-blocks/src/utils.ts

```diff
@@ -113,69 +116,82 @@ const addBlockProperties = block => {
 		block.version = 0;
 	}
 	if (block.numberOfTransactions === undefined) {
-		if (block.transactions === undefined) {
-			block.numberOfTransactions = 0;
-		} else {
-			block.numberOfTransactions = block.transactions.length;
-		}
+		block.numberOfTransactions =
+			block.transactions === undefined ? 0 : block.transactions.length;
 	}
 	if (block.payloadLength === undefined) {
 		block.payloadLength = 0;
 	}
 	if (block.transactions === undefined) {
 		block.transactions = [];
 	}
+
 	return block;
 };
 
-const deleteBlockProperties = block => {
+export const deleteBlockProperties = (block: BlockHeader) => {
 	const reducedBlock = {
 		...block,
 	};
+	/* tslint:disable:no-delete */
 	if (reducedBlock.version === 0) {
 		delete reducedBlock.version;
 	}
-	// verifyBlock ensures numberOfTransactions is transactions.length
+	// VerifyBlock ensures numberOfTransactions is transactions.length
 	if (typeof reducedBlock.numberOfTransactions === 'number') {
 		delete reducedBlock.numberOfTransactions;
 	}
-	if (reducedBlock.totalAmount.equals(0)) {
+	if (reducedBlock.totalAmount.eq(0)) {
 		delete reducedBlock.totalAmount;
 	}
-	if (reducedBlock.totalFee.equals(0)) {
+	if (reducedBlock.totalFee.eq(0)) {
 		delete reducedBlock.totalFee;
 	}
 	if (reducedBlock.payloadLength === 0) {
 		delete reducedBlock.payloadLength;
 	}
-	if (reducedBlock.reward.equals(0)) {
+	if (reducedBlock.reward.eq(0)) {
 		delete reducedBlock.reward;
 	}
 	if (reducedBlock.transactions && reducedBlock.transactions.length === 0) {
```

Comment:
`(reducedBlock?.transactions.length === 0)`

Reply:
For this PR, im not doing any refactoring if not necessary by typescript.
We will refactor the code on the next iteration